src/int.rs

```diff
@@ -0,0 +1,470 @@
+//! Stack-allocated big signed integers.
+
+use core::fmt;
+
+use num_traits::ConstZero;
+use subtle::{Choice, ConditionallySelectable, ConstantTimeEq, ConstantTimeGreater, CtOption};
+
+use crate::{
+    Bounded, ConstChoice, ConstCtOption, ConstantTimeSelect, Encoding, Limb, NonZero, Odd, Uint,
+    Word,
+};
+
+#[macro_use]
+mod macros;
+
+mod add;
+mod cmp;
+mod div;
+mod encoding;
+mod expand;
+mod mul;
+mod sub;
+
+#[cfg(feature = "rand_core")]
+mod rand;
+
+/// Stack-allocated big _signed_ integer.
+/// See [`Uint`] for _unsigned_ integers.
+///
+/// Created as a [`Uint`] newtype.
+#[allow(clippy::derived_hash_with_manual_eq)]
+#[derive(Copy, Clone, Hash)]
+pub struct Int<const LIMBS: usize>(Uint<LIMBS>);
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// The value `0`.
+    pub const ZERO: Self = Self(Uint::ZERO); // Bit sequence (be): 0000....0000
+
+    /// The value `1`.
+    pub const ONE: Self = Self(Uint::ONE); // Bit sequence (be): 0000....0001
+
+    /// The value `-1`
+    pub const MINUS_ONE: Self = Self::FULL_MASK; // Bit sequence (be): 1111....1111
+
+    /// Smallest value this [`Int`] can express.
+    pub const MIN: Self = Self(Uint::MAX.bitxor(&Uint::MAX.shr(1u32))); // Bit sequence (be): 1000....0000
+
+    /// Maximum value this [`Int`] can express.
+    pub const MAX: Self = Self(Uint::MAX.shr(1u32)); // Bit sequence (be): 0111....1111
+
+    /// Bit mask for the sign bit of this [`Int`].
+    pub const SIGN_BIT_MASK: Self = Self::MIN; // Bit sequence (be): 1000....0000
```

Comment:
for consistency either rename it to `SIGN_MASK` or rename `FULL_MASK` to `FULL_BIT_MASK`. I prefer the former.

Reply:
done!